mongo/mongo_test.go

```diff
@@ -617,38 +639,40 @@ func (s *MongoSuite) TestInstallMongodServiceExists(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 
 	c.Check(s.data.Installed(), gc.HasLen, 0)
-	s.data.CheckCallNames(c, "Installed", "Exists", "Running")
+	s.data.CheckCallNames(c, "Running")
 }
 
 func (s *MongoSuite) TestNewServiceWithReplSet(c *gc.C) {
-	conf := mongo.NewConf(s.makeConfigArgs(c.MkDir()))
+	confArgs := s.makeConfigArgs(c.MkDir())
+	conf := mongo.NewConf(&confArgs)
```

Comment:
Is there a reason why this was changed?
It's idiomatic to pass in smallish structs  to methods that don't modify the data